lib/index.ts

```diff
@@ -99,24 +99,53 @@ export default class Backburner {
     this._onBegin = this.options.onBegin || noop;
     this._onEnd = this.options.onEnd || noop;
 
+    this._boundRunExpiredTimers = () => {
+      this._runExpiredTimers();
+    };
+
+    this._boundAutorunEnd = () => {
+      // if the autorun was already flushed, do nothing
+      if (this._autorun === null) { return; }
+
+      this._autorun = null;
+      this.end(true /* fromAutorun */);
+    };
+
     let _platform = this.options._platform || {};
     let platform = Object.create(null);
     platform.setTimeout = _platform.setTimeout || ((fn, ms) => setTimeout(fn, ms));
     platform.clearTimeout = _platform.clearTimeout || ((id) => clearTimeout(id));
-    platform.next = _platform.next || ((fn) => SET_TIMEOUT(fn, 0));
     platform.clearNext = _platform.clearNext || platform.clearTimeout;
     platform.now = _platform.now || (() => Date.now());
 
-    this._platform = platform;
+    // TODO: do we need this?
+    if (_platform.next) {
```

Comment:
should we make this configurable, and just plug in RSVP's asap in ember? Then we avoid duplicate code

Reply:
We could make asap also use native Promise for its asap.

Reply:
@stefanpenner - I think we want to avoid the extra level of queuing that RSVP.asap does. Specifically we need to leverage the fact that microtasks that we enqueue are in the correct position (e.g. interleaving of native promises and RSVP.Promises should maintain the same resolve ordering).